Yank/repex.py

```diff
@@ -222,7 +222,34 @@ def _are_subfiles_open(self):
                 open_check_list.append(storage.isopen())
         return np.all(open_check_list)
 
-    def open(self, mode='r', convention='ReplicaExchange'):
+    def _ensure_dimension_exists(self, dimname, dimsize):
+        """
+        Ensure a dimension exists and is of the appropriate size,
+        creating it if it does not already exist.
+
+        A ``ValueError`` is raised if ``dimsize`` does not match the existing dimension size.
+
+        Parameters
+        ----------
+        dimname : str
+            The dimension name
+        dimsize : int
+            The dimension size
+
+        """
+        if dimname not in self._storage_analysis.dimensions:
+            self._storage_analysis.createDimension(dimname, dimsize)
+        else:
+            # Check dimension matches expected size
+            dimension = self._storage_analysis.dimensions[dimname]
+            if dimsize == 0:
+                if not dimension.isunlimited():
+                    raise ValueError("NetCDF dimension '%s' already exists: was previously unlimited, but tried to declare it with size %d" % (dimension.name, dimsize))
```

Comment:
My only request is to use the Python `.format()` syntax instead of `%` syntax for string formatting. Really minor thing but otherwise is fine!

Reply:
I hate `.format()`.